service/src/java/org/apache/hive/service/auth/ldap/KerberosLdapFilterEnforcer.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hive.service.auth.ldap;
+
+import java.io.IOException;
+
+import javax.security.sasl.AuthenticationException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.hive.conf.HiveConf;
+
+/**
+ * Singleton helper that applies configured LDAP filters for Kerberos-authenticated users.
+ */
+public final class KerberosLdapFilterEnforcer {
```

Comment:
Personally I would like this class exposing the only `applyLdapFilter(hiveconf, username)` method, and holding all related to the Ldap authentication, initialization, like:

```
private KerberosLdapFilterEnforcer(HiveCon conf, DirSearchFactory factory) {
...
}
// authentication, 
boolean applyLdapFilter(String userName) {
...
}

```